File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-10.java
class Solution {
    public int search(int[] nums, int target) {
        //[[4,5,6,7m],[0,1,2]]
        //[[4,5,6],[0m,1,2,3]]
        //[[4,5,][1,2,3]]
        //target < l  -> l = mid + 1
        //target > l 
        int l = 0, r = nums.length - 1;

        while(l <= r) {
            int mid = l + (r - l) / 2;
            if(nums[mid] == target) return mid;
            
            if(nums[l] <= nums[mid]) {
                //(3, 5]
                if(nums[l] <= target && target < nums[mid]) {
                    r = mid - 1;
                } else {
                    l = mid + 1;
                }
            } else {
                //[1, 2)
                if(nums[r] >= target && target > nums[mid]) {
                    l = mid + 1;
                } else {
                    r = mid - 1;
                }
            }
        }


        return -1;
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-10.cs
public class Solution {
    public int[] TwoSum(int[] numbers, int target) {
        int l = 0; int r = numbers.Length - 1;

        for(int i = 0; i < numbers.Length; i++) {
            int sum = numbers[l] + numbers[r];

            if(sum < target) l++;
            else if(sum > target) r--;
            else return new int[] {l+1, r+1};
        }

        return new int[] {-1, -1};
    }
}
